miio/device.py

```diff
@@ -155,6 +161,32 @@ def _fetch_info(self) -> DeviceInfo:
                 "Unable to request miIO.info from the device"
             ) from ex
 
+    def _initialize_descriptors(self) -> None:
+        """Cache all the descriptors once on the first call."""
+        if self._sensors is not None:
+            return
+
+        status = self.status()
+
+        # Sensors
+        self._sensors = status.sensors()
+
+        # Settings
+        self._settings = status.settings()
+        for setting in self._settings.values():
+            if setting.setter is None:
+                if setting.setter_name is None:
```

Comment:
This is rather deep conditional, any way to simplify?

Reply:
I managed to simplify it a bit to be one level less.